imports/lib/schemas/typedSchemas.ts

```diff
@@ -30,6 +30,7 @@ type StringOverrides<T> = T extends string ? {
   exclusiveMin?: boolean;
   exclusiveMax?: boolean;
   regEx?: RegExp | RegExp[];
+  custom?: SchemaDefinition['custom'];
```

Comment:
Does this only apply to string or should it be part of `SharedOverrides`?

Reply:
Huh.  I think I don't fully understand how `FieldOverrides` works -- to me it reads like the sum type of a variety of other types, so if I had a particular concrete type T that extended string, then FieldOverrides<T> would be:

```
FieldOverrides<T> =
  Record<string, never> | // NumberOverrides<T> |
  Record<string, never> | // DateOverrides<T> |
  Record<string, never> | // ArrayOverrides<T> |
  StringOverrides<T> |
  Record<string, never> | // ObjectOverrides<T> |
  SharedOverrides<T>;
```

which means that the overrides for that field can include fields from `StringOverrides`, or things from `SharedOverrides`, but not both -- the type must fit into one of the two outlines.

In practice this works out; the only times we specify `custom` we specify no other overrides, so as written in this PR, the type is inferred as `StringOverrides<T>`.  If I moved `custom?` to `SharedOverrides<T>`, I imagine that the type of a field override would be inferred as matching `SharedOverrides<T>` instead. 

Maybe we actually meant for this to be more like:

```
export type FieldOverrides<T> = (
  NumberOverrides<T> |
  DateOverrides<T> |
  ArrayOverrides<T> |
  StringOverrides<T> |
  ObjectOverrides<T>) &
  SharedOverrides<T>;
```

instead, where we'd accept the additional SharedOverrides specifiers with any type, and (when the type matched) also allow keys from the more-specific `*Overrides` type?  That'd align more with my guess of what these types were trying to model, but also some other things fail to typecheck if we do that.

I can move `custom` to `SharedOverrides` though; it doesn't need to be bound to only be usable on strings.